lib/phx_tools_web/live/phx_tools_components.ex

```diff
@@ -0,0 +1,273 @@
+                Phx.tools is a shell script for Linux and macOS that configures the development environment for you in a few easy steps. Once you finish running the script, you'll be able to start the database server, create a new Phoenix application, and launch the server.
+              </p>
+            </div>
+            <div
+              class="mx-auto"
+              data-asciicast={
+                if @live_action == :macOS,
+                  do: "bJMOlPe5F4mFLY0Rl6fiJSOp3",
+                  else: "XhDpRstBJ4df2gfiRfp0awDPO"
+              }
+              id={"asciinema-#{Utils.random_id()}"}
+              phx-hook="AsciinemaHook"
+            >
+            </div>
+
+            <p class="md:my-5 text-white md:px-12 text-xs md:text-sm lg:text-base font-martian">
+              Read about website updates here -
+              <a
+                href="https://optimum.ba/blog/exciting-updates-to-phx-tools"
+                class="text-[#24B2FF] font-martian text-sm underline"
+                target="_blank"
+              >
+                https://optimum.ba/blog/exciting-updates-to-phx-tools
+              </a>
+            </p>
+          </div>
+        </div>
+      </div>
+      <.footer />
+    </div>
+    """
+  end
+
+  attr :installation_instructions, :list
+  attr :live_action, :atom
+  attr :operating_system, :string
+
+  slot :inner_block, required: true
+
+  @spec os_instructions(assigns()) :: rendered()
+  def os_instructions(assigns) do
+    ~H"""
+    <div
+      id="installation-instructions"
+      phx-click-away={JS.hide(to: "#installation-instructions")}
+      class="hidden font-normal solved-height font-martian bg-[#110A33] absolute md:w-[90%]"
+    >
+      <div class="block">
+        <div class="h-full shadow-custom shadow-md rounded-md pb-2">
+          <div class="text-start px-[3%] lg:text-xl md:text-lg sm:text-md">
+            <h1 class="text-white text-center text-sm md:text-base lg:text-lg lg:my-[5%] md:my-[2%] sm:my-[2%] lg:pt-5">
+              <%= Index.capitalize_os_name_first_letter("#{@live_action}") %> installation process
+            </h1>
+            <ol class="list-decimal ml-3 pl-5 text-xs md:text-sm lg:text-base text-white lg:mt-4 sm:mt-2 leading-6">
+              <%= for instruction <- @installation_instructions do %>
+                <li class="mb-2">
+                  <%= render_slot(@inner_block, instruction) %>
+                </li>
+              <% end %>
+            </ol>
+          </div>
+        </div>
+      </div>
+    </div>
+    """
+  end
+
+  attr :class, :string
+  attr :live_action, :atom, required: true
+  attr :operating_system, :string, required: true
+
+  @spec command_select_button(assigns()) :: rendered()
+  def command_select_button(assigns) do
+    ~H"""
+    <.link href={if Atom.to_string(@live_action) == "macOS", do: ~p"/linux", else: ~p"/macOS"}>
```

Comment:
Again, a lot of ifs in this function could be just pattern matching inside some smaller function components.